Packs/ServiceDeskPlus/Integrations/ServiceDeskPlus/ServiceDeskPlus.py

```diff
@@ -0,0 +1,853 @@
+        Demisto Outputs.
+    """
+    request_id = args.get('request_id')
+    result = client.http_request('GET', url_suffix=f'requests/{request_id}/resolutions')
+
+    context: dict = defaultdict(list)
+    output = create_output(result.get('resolution', {}))
+    hr = {}
+    if output:
+        context['ServiceDeskPlus.Request(val.ID===obj.ID)'] = {
+            'Resolution': output
+        }
+        hr = resolution_human_readable(output)
+    markdown = tableToMarkdown(f'Resolution of request {request_id}', t=hr)
+    return markdown, context, result
+
+
+def close_request_command(client: Client, args: dict) -> Tuple[str, dict, Any]:
+    """
+    Close the request with the given request_id with comments and resolution defined by the user.
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+
+    Returns:
+        Demisto Outputs.
+    """
+    request_id = args.get('request_id')
+    params = {
+        "request": {
+            "closure_info": {
+                "requester_ack_resolution": args.get('requester_ack_resolution'),
+                "requester_ack_comments": args.get('requester_ack_comments'),
+                "closure_comments": args.get('closure_comments'),
+                "closure_code": {
+                    'name': args.get('closure_code')
+                }
+            }
+        }
+    }
+    result = client.http_request('PUT', url_suffix=f'requests/{request_id}/close', params=params)
+    hr = f'### Successfully closed request {request_id}'
+    return hr, {}, result
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_limit: int, status: str, fetch_filter: str) -> list:
+    date_format = '%Y-%m-%dT%H:%M:%S'
+    last_run = demisto.getLastRun()
+    if not last_run:  # if first time running
+        new_last_run = {
+            'time': date_to_timestamp(parse_date_range(fetch_time, date_format=date_format, utc=False)[0])
+        }
+    else:
+        new_last_run = last_run
+    demisto_incidents: List = list()
+    time_from = new_last_run.get('time')
+    time_to = date_to_timestamp(datetime.now(), date_format=date_format)
+    list_info = create_fetch_list_info(str(time_from), str(time_to), status, fetch_filter, fetch_limit + 1)
+    params = {
+        'input_data': f'{list_info}'
+    }
+
+    # Get incidents from Service Desk Plus
+    demisto.info(f'Fetching ServiceDeskPlus incidents. with the query params: {str(params)}')
+
+    incidents = client.get_requests(params=params).get('requests', [])
+
+    if incidents:
+        count = 0
+        last_run_id = last_run.get('id', '0')
+        last_incident_id = last_run.get('id', '0')
+        cur_time = new_last_run.get('time', 0)
+        incident_creation_time = new_last_run.get('time', 0)
+
+        for incident in incidents:
+            if count >= fetch_limit:
+                break
+            # Prevent fetching twice the same incident
+            if incident.get('id') == last_run_id:
```

Comment:
shouldn't we break here?

Reply:
No, the last incident that was previously fetched might be the first incident that will be returned the next time fetch is called. We want to skip this incident and use all others so we can't break here.